### PR TITLE
Rename create and insert

### PR DESCRIPTION
**Pull request by SferaDev**
```ts
  /*
   * Creates a single record in the table.
   * @param object Object containing the column names with their values to be stored in the table.
   * @returns The full persisted record.
   */
  abstract create(object: Data & Partial<Identifiable>): Promise<Record>;

  /**
   * Creates a single record in the table with a unique id.
   * @param id The unique id.
   * @param object Object containing the column names with their values to be stored in the table.
   * @returns The full persisted record.
   */
  abstract create(id: string, object: Data): Promise<Record>;
```

### CODE REVIEW

File: client/src/schema/repository.ts [hunk 1]
```diff
@@ -132,7 +133,23 @@ export class RestRepository<Data extends BaseData, Record extends XataRecord = D
...
 
-  async create(object: Data): Promise<Record> {
+  async create(object: Data): Promise<Record>;
+  async create(recordId: string, object: Data): Promise<Record>;
+  async create(a: string | Data, b?: Data): Promise<Record> {
+    if (isString(a) && isObject(b)) {
+      if (a === '') throw new Error("The id can't be empty");
+      return this.#insertRecordWithId(a, b);
+    } else if (isObject(a) && isString(a.id)) {
+      if (a.id === '') throw new Error("The id can't be empty");
```
Comment (gimenete) [inline client/src/schema/repository.ts:143]:
```suggestion
      if (!a.id) throw new Error("The id can't be empty, null or undefined");
```

The test will need to be updated.

File: test/integration.test.ts [hunk 2]
```diff
@@ -532,3 +460,104 @@ describe('integration tests', () => {
...
+  });
+
+  test('create user with empty inline id is not allowed', async () => {
+    expect(
+      client.db.users.create({
+        id: '',
+        full_name: 'John Doe 3',
+        email: 'john3@doe.com'
+      })
+    ).rejects.toMatchInlineSnapshot(`[Error: The id can't be empty]`);
```
Comment (gimenete) [inline test/integration.test.ts:551]:
```suggestion
    ).rejects.toMatchInlineSnapshot(`[Error: The id can't be empty, null or undefined]`);
```

**Review by gimenete — APPROVED** [review]

File: client/src/schema/repository.ts (hunk 1 above)
Comment (SferaDev) [reply client/src/schema/repository.ts:143]: `isString(a.id)` checks for null and undefined already

File: test/integration.test.ts (hunk 2 above)
Comment (SferaDev) [reply test/integration.test.ts:551]: If null or undefined the error "[Error: Invalid arguments for create method]". We already have a test for it: https://github.com/xataio/client-ts/pull/136/files#diff-8fdaf3b73054c861d4b1afc72e76086d85c318c75bde88a452ef344c24b7d8b0R554-R563

File: client/src/schema/repository.ts (hunk 1 above)
Comment (gimenete) [reply client/src/schema/repository.ts:143]: Cool

**Pull request merged**